server/src/main/java/org/opensearch/cluster/metadata/Metadata.java

```diff
@@ -1329,6 +1336,36 @@ public Builder removeDataStream(String name) {
             return this;
         }
 
+        public Builder resourceLimitGroups(final Map<String, ResourceLimitGroup> resourceLimitGroups) {
```

Comment:
nit: 
```suggestion
        public Builder putResourceLimitGroups(final Map<String, ResourceLimitGroup> resourceLimitGroups) {
```

Reply:
I have kept these methods inline with rest of the methods in class.